src/internet_identity/src/archive.rs

```diff
@@ -14,27 +14,40 @@ use internet_identity_interface::*;
 use serde_bytes::ByteBuf;
 use sha2::Digest;
 use sha2::Sha256;
+use std::rc::Rc;
 use std::time::Duration;
-use ArchiveState::{Created, CreationInProgress, NotCreated};
+use ArchiveState::{Configured, Created, CreationInProgress, NotConfigured};
 use CanisterInstallMode::Upgrade;
 
-#[derive(Clone, Debug, Default, CandidType, Deserialize, Eq, PartialEq)]
-pub struct ArchiveInfo {
-    pub expected_module_hash: Option<[u8; 32]>,
-    pub state: ArchiveState,
-}
-
 /// State of the archive canister.
 #[derive(Eq, PartialEq, Clone, CandidType, Debug, Deserialize)]
 pub enum ArchiveState {
-    NotCreated,
-    CreationInProgress(Timestamp), // timestamp when creation was initiated
-    Created(ArchiveData),
+    /// Not configured and not created.
+    NotConfigured,
+    /// Configured but not created.
+    Configured {
+        /// Archive related configuration
+        config: ArchiveConfig,
+    },
```

Comment:
What is configured? The II persistent state or an existing archive canister? Also I'm a little confused why this is needed

Reply:
I switched around the types here to make it more consistent. I.e. having an archive now implies having a configuration as well.

And to represent (no config, no archive) and (config, no archive) I had to introduce a new state. `Configured` represents the (config, no archive) state.

Reply:
Sorry, the question wasn't clear. I was asked "by _configured_, do you mean that the archive is created and configured or that we have a config configured in the II canister", and I know understand you mean the latter.